### DISCUSSION TITLE
Clang Build Analyzer - August 2024

### DISCUSSION THREAD

**Discussion by nick-l-o3de** [o3de/o3de#18196] [Developer Articles]
Please see attached, 2 files.  Building a small project
[clang_build_analysis_non_unity.txt](https://github.com/user-attachments/files/16511958/clang_build_analysis_non_unity.txt)

[clang_build_analysis_unity.txt](https://github.com/user-attachments/files/16511960/clang_build_analysis_unity.txt)

 from scratch with, and without unity mode enabled.

Analysis follows...
Update:  Here's an Include-What-You-Use for AzCore.  

[iwyu_azcore_verbosity_1_nonunity.txt](https://github.com/user-attachments/files/16531305/iwyu_azcore_verbosity_1_nonunity.txt)

**Comment by spham-amzn** [comment]
Looks like the biggest offender is from the code-gen thats happening in Script Canvas.

**Reply by nick-l-o3de** [reply]
That's always been the case, but I'm surprised to see the statistical profiler in there too...

**Comment by pollend** [comment]
The ScriptCanvas MainWindows has a lot of includes. wonder how many of those includes are needed.

**Reply by nick-l-o3de** [reply]
I can run a IncludeWhatYouUse run on ScriptCanvas main windows, it just takes like, literally hours to get there.  It makes a txt file thats incredibly large
